qa/tasks/cephfs/caps_helper.py

```diff
@@ -216,29 +206,34 @@ def run_mon_cap_tests(self, def_fs, client_id):
         else:
             raise RuntimeError('run_mon_cap_tests(): mon cap not found in '
                                'keyring. keyring -\n' + keyring)
+        return moncap
 
-        keyring_path = def_fs.admin_remote.mktemp(data=keyring)
+    def run_mon_cap_tests(self, fs, client_id):
+        """
+        Check that MON cap is enforced for a client by searching for a Ceph
+        FS name in output of cmd "fs ls" executed with that client's caps.
+
+        fs is any fs objcet so that ceph commands can be exceuted.
+        """
+        get_cluster_cmd_op = fs.mon_manager.raw_cluster_cmd
+        keyring = get_cluster_cmd_op(args=f'auth get client.{client_id}')
```

Comment:
You could pass `client_id` to `_get_mon_cap_from_keyring` and let it fetch the keyring.

Reply:
No. That's a bad idea because keyring is needed in the code below this line.

Reply:
Never mind then. It possibly still could be improved, but its not a deal breaker.